content/blog/2025-05-13-SCIF.md

```diff
@@ -0,0 +1,116 @@
+
+Previously, SCIF had no public reference manual for users. We introduce CI to build and publish a public language reference manual. Additionally, SCIF had no sanity checkers for contributors. We introduce GitHub actions to verify compilation and running of the compiler.
+
+Finally, SCIF is currently only runnable through Gradle. This requires users to checkout the repository (and submodule), install Java and Gradle, and understand how to set up the repository. This seriously hinders the usability of SCIF as a language, as most users of a compiler simply want to run it. We begin work to untangle hardcoded paths to the local filesystem and package the compiler as a reusable, compiled JAR for distribution. 
+<!-- We additionally are working to improve the build and run time of the compiler. We've began to see small improvements from our better integration of SHErrLoc, andNumerous builtin contracts are recompiled on every execution, and SHErrLoc ShAdditionally, builtin contract files are recompiled every time the compiler runs, which is un -->
+The work here is still ongoing, but this will be a top priority and integral to larger SCIF project's continued success.
+
+## Performance, Results and Testing
+
+For multi-contract supports, we wrote 12 tests, ranging from the most simple multi-contract structure, to real-world applications like `Uniswap` with >400 LOC and containing multiple contracts. We also tested complicated import relationship and it compiles successfully without errors. For example,
+
+```
+[File 1]
+import "file2.scif";
+import "file3.scif";
+interface B {...}
+contract A {...uses B, C, D, E...}
+
+[File 2]
+import "file4.scif";
+interface C {...uses E...}
+
+[File 3]
+interface D {...}
+
+[File 4]
+interface E {...}
+```
+
+This thing would compile. However, if interface `B` uses `A`, interface `C` uses `B`, or interface `C` uses `D`, the compilation will fail. 
+
+## Challenges
+
+Our two biggest challenges can both generally be summarized as having needed to work in a time-crunch, due to changing our project track later in the game; and the difficulties of adding on optimizations onto a codebase which was not entirely our own. We were able to get over the time hurdle by readjusting the scope of the project and working concurrently in a couple of sprints; it was also helpful to divide up tasks based on expertise and familiarity with the SCIF project. Working wtih the codebase was challenging due to the somewhat sparse documentation in places; to this effect, we've added as a part of our goal with this compiler to improve overall documentation of methods going forward.
```

Comment:
> adding on optimizations

I didn't see any actual _optimizations_ here. Do you maybe mean, like, _changes_, more generically?